spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -49,38 +65,32 @@ public MessageChannel resolveDestination(String name) {
 		}
 		catch (DestinationResolutionException e) {
 		}
-		if (name.contains(":")) {
-			if (binderFactory != null) {
-				String[] tokens = name.split(":", 2);
+		synchronized (this) {
+			try {
+				return super.resolveDestination(name);
+			}
+			catch (DestinationResolutionException e) {
+			}
+			if (this.beanFactory != null && this.binderFactory != null) {
+				channel = new DirectChannel();
+				this.beanFactory.registerSingleton(name, channel);
+				channel = (MessageChannel) this.beanFactory.initializeBean(channel, name);
 				String transport = null;
-				String type;
-				if (tokens.length == 2) {
-					type = tokens[0];
-				}
-				else if (tokens.length == 3) {
-					transport = tokens[0];
-					type = tokens[1];
-				}
-				else {
-					throw new IllegalArgumentException("Unrecognized channel naming scheme: " + name + " , should be" +
-							" [<transport>:]<type>:<name>");
+				if (name.contains(":")) {
```

Comment:
Maybe we can use a different character here and get done with the colon? Indirectly, this is a concern for Kafka, where the colon is a legal character.